libs/wingsdk/src/shared-gcp/bucket.inflight.ts

```diff
@@ -54,9 +54,14 @@ export class BucketClient implements IBucketClient {
    * @throws if `srcKey` object doesn't exist or if it matches `dstKey`.
    */
   public async rename(srcKey: string, dstKey: string): Promise<void> {
-    return Promise.reject(
-      `rename is not implemented: (srcKey=${srcKey}, dstKey=${dstKey})`
-    );
+    if (srcKey === dstKey) {
+      throw new Error(
+        `Renaming an object to its current name is not a valid operation (srcKey=${srcKey}, dstKey=${dstKey}).`
+      );
+    }
+
+    await this.copy(srcKey, dstKey);
+    await this.delete(srcKey);
```

Comment:
Do we need to put it in a try-catch? 
@Chriscbr WDYT?

Reply:
Why would you do that? You already check that the object exists so the copy operation cannot fail. You could call `tryDelete()` instead of `delete()` if you wanted, but that seems unecessary.

Reply:
Why would you do that? You already check that the object exists so the copy method cannot fail. You could call `tryDelete()` instead of `delete()` if you wanted, but I think that you should actually throw the error.